README.md

```diff
@@ -1,52 +1,105 @@
+users. Other values will be interpretted as a username and the role will
+modify per-user configuration stored under `~/.ssh/config` of the given user.
+The user needs to exists before invoking this role otherwise it will fail.
 
-```yaml
-ssh_foo: "oof"
-ssh_bar: "baz"
-```
+ * `ssh_skip_defaults`:
+
+By default (`auto`), the role writes the system-wide configuration file
+`/etc/ssh/ssh_config` and keeps OS defaults defined there (*true*). This is
+automatically disabled, when a drop-in configuration file is created
+(`ssh_drop_in_name!=null`) or when per-user configuration file is created
+(`ssh_user!=null`).
+
+ * `ssh_drop_in_name`:
+
+This defines the name for the drop-in configuration file to be placed in
+system-wide drop-in directory. The name is used in the template
+defined by `__ssh_drop_in_template` (by default
+`/etc/ssh/ssh_config.d/{name}.conf`) to reference the configuration file
+to be modified. If the system does not support drop-in directory, setting
+this option will make the play fail. Default is `null` if the system does
+not support drop in directory and `00-ansible` otherwise.
+
+The suggested format is `NN-name`, where `NN` is two-digit number used for
+sorting the and `name` is any descriptive name for the content or the owner
+of the file.
+
+ * `ssh`:
+
+A dict containing configuration options and respective values. See example
+below.
+
+ * `ssh_...`:
+
+Simple variables consisting of the option name prefixed with `ssh_` can be
+used rather than a dict above. The simple variable overrides values in dict
+above.
 
-### Variables Exported by the Role
+ * `ssh_additional_packages`:
 
-This section is optional.  Some roles may export variables for playbooks to
-use later.  These are analogous to "return values" in Ansible modules.  For
-example, if a role performs some action that will require a system reboot, but
-the user wants to defer the reboot, the role might set a variable like
-`ssh_reboot_needed: true` that the playbook can use to reboot at a more
-convenient time.
+This role automatically installs packages needed for most common use cases
+on given platform. If some additional packages need to be installed (for
+example `openssh-keysign` for hostbased authentication), they can be specified
+in this variable.
 
-Example:
+### Secondary role variables:
+
+These variables are used by the role internals and can be used to override
+the defaults that correspond to each supported platform.
+
+ * `__ssh_packages`:
+
+Minimal list of packages to install on a given platform for openssh clients
+to function. Do not override this variable if you need to install additional
+packages. Use `ssh_additional_packages` instead.
+
+ * `ssh_config_file`:
+
+The configuration file that will be written by this role. The default is
+defined by template `__ssh_drop_in_template` if system has drop-in directory
+or `/etc/ssh/ssh_config` otherwise. If `ssh_user!=null`, the
+default is `~/.ssh/config`.
+
+ * `__drop_in_template`:
```

Comment:
I cannot find `__drop_in_template` in the source code. Could it be `__ssh_drop_in_template`? 

And you cannot modify `__ssh_drop_in_template`, I'd think?

Reply:
right. It was a typo. Thanks.